jsonnet/components/cluster-monitoring-operator.libsonnet

```diff
@@ -330,11 +330,18 @@ function(params) {
     metadata: {
       name: 'cluster-monitoring-view',
     },
-    rules: [{
-      apiGroups: [''],
-      resources: ['namespaces'],
-      verbs: ['get'],
-    }],
+    rules: [
+      {
+        apiGroups: [''],
+        resources: ['namespaces'],
+        verbs: ['get'],
+      },
+      {
+        apiGroups: ['monitoring.coreos.com'],
+        resources: ['prometheuses/api'],
+        verbs: ['get', 'create', 'update'],
+      },
+    ],
   },
 
```

Comment:
I would create a new role (not clusterrole) like we have for alertmanager-user-workload (e.g. `cluster-monitoring-api-reader`) so users can migrate to the new role instead of granting a clusterrole.

Reply:
Good idea, will use a new role for thanos/prometheus api access.

Reply:
The role `cluster-monitoring-api-reader` is added into  `assets/cluster-monitoring-operator/cluster-monitoring-api-reader-role.yaml`. 
After we replace Oauth proxy in CMO, this role will replace `cluster-monitoring-view` in rolebindings.